src/modules/referral/services/service.ts

```diff
@@ -186,7 +186,13 @@ export class ReferralService {
   }
 
   public extractReferralAddressUsingDelimiter(data: string) {
-    const referralData = this.subtractFunctionArgsFromCallData(data);
+    let referralData = "";
+
+    try {
+      referralData = this.subtractFunctionArgsFromCallData(data);
```

Comment:
Do you think it'd be better to explicitly check the `data`'s length rather than the more vague try-catch which might catch a different error that we don't want to catch?

Reply:
I don't think we can check the data's length because it can be any length, right? You can attach any arbitrary data to the calldata